### PR TITLE
Fix relative paths in up-to-date checker

### PR DESCRIPTION
**Pull request by panopticoncentral**
**Customer scenario**

If a project CopyIfNewer file has a relative path in an item that places it in a subfolder (i.e. "foo\bar.txt"), then the up-to-date checker will always fail that project. The problem is that it doesn't look at the relative paths, only the file name (so it looks for "bar.txt" in the output folder, not "foo\bar.txt").

**Bugs this fixes:** 

#3023 

**Workarounds, if any**

User has to always build, or remove the file.

**Risk**

Low, this just fixes a situation that doesn't work today.

**Performance impact**

None, we're still checking the same number of things as before.

**Is this a regression from a previous update?**

No.

**How was the bug found?**

Customer report

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs [hunk 1]
```diff
@@ -481,13 +481,21 @@ private bool CheckCopyToOutputDirectoryFiles(BuildUpToDateCheckLogger logger, ID
...
             {
-                var filename = item.Link ?? Path.GetFileName(item.Path);
+                var filename = string.IsNullOrEmpty(item.Link) ? item.Path : item.Link;
 
                 if (string.IsNullOrEmpty(filename))
                 {
                     continue;
                 }
 
+                if (Path.IsPathRooted(filename))
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs]: UnconfiguredProject.MakeRelative and it handles `C:Foo.txt`.

**Comment by panopticoncentral** [comment]
@davkean How does that look?

**Comment by panopticoncentral** [comment]
@davkean Yes, I'll handle that in #2452.

**Comment by gulbanana** [comment]
thanks for improving this! it’ll be a big help

**Pull request merged**